### ISSUE TITLE
[a11y Staging Review] IIR, Welcome Pathway

### ISSUE THREAD

**Issue by coforma-jamie** [department-of-veterans-affairs/va.gov-team#95851]
# CAIA A11y Support

> [!NOTE]
> - **Review:** **Coded product** - to be ready week of 11/25/2024
> - **CAIA Review Date**: `11/27/2024` 
> - **Review Purpose**: Staging Prep
> - **To Be Reviewed By**: @mutiaadhocteam 

## Next Steps

> [!TIP]
> **DEFINITION OF DONE**: Refer to the task list in this ticket, for actionable items.

## Description

The team is currently developing a Figma prototype, that will be referenced and used when creating the coded prototype for user research. 

### Timeframe
Week of 11/~11~ 25 

### No Accessibility Specialist on Team
CAIA accessibility specialists are available to provide support to IIR through this process with accessibility support, as this team does not have an accessibility specialist. 

### Provided via CAIA Intake

- [Product outline](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/veteran-onboarding/product-outline.md)
- [User flows](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1715019181472/206d1983971576ee7696c1984666185a2fbff545?fromVisitorModal=true&sender=u4356f0478274fc3cca5b5144)
- Staging link `tbd`

## CAIA Assignee(s)

<details><summary>Toggle to view assignee info</summary>
- `tbd`

</details>

## Team Info

<details><summary>Toggle to view team info</summary>

- Team name: IIR (Iterate, Innovate, Run)
- OCTO-DE product owner: Jeff Barns
- Product manager: Pete Egan
- Designer: Liza McRuer/Raquel Eisele
- FE engineer: John Rodriguez/Nick Sayre
- Product/team Slack channel: #iir-product-teams-public

</details>

## Definition of Done

```[tasklist]
### A11y Tasks
- [x] Receives link to staging environment
- [x] Receives test user account information if applicable
- [x] Creates an audit document encompassing all widget states and placement
- [x] Audits
- [x] Shares findings with IIR team
```

```[tasklist]
## Ticket Updates (CAIA Internal)
- [x] _Connect to an `Epic` or `Intake` (what body of work is this a part of?)_
- [x] _Label with `Originator/Team` (product team or stakeholder requesting support)_
- [x] _Label date in the `Open Date` field_
- [x] _Label with `Estimate` (level of effort expected for this ticket)_
- [x] _Add `Assignee(s)` name(s) to ticket_
- [x] _Add `Assignee(s) name(s) to each task` they will complete via handle tag (if known)_
- [x] _Select a `Priority Level`_
- [x] _Update date in `Last Checked` field_
- [x] _Label with `Actual` (level of effort it took to complete this ticket)_
- [x] _Update date in_ `Closed  Date`
```

**Comment by sara-amanda** [comment]
## Timeframe Shift 
- 2 weeks until pilots
- Week of November 25 for pilots (TBD)
- Will update Capacity Calendar Accordingly

### Slack Update
`iir-product-teams-public`

> Raquel Eisele [Yesterday at 5:28 PM](https://dsva.slack.com/archives/C05RJS5DANT/p1731450522330639?thread_ts=1731425468.647049)
> Hey Jamie, I just had a conversation with my dev and it sounds like sadly we are not ready to conduct the pilot this week. If we implement the changes you highlighted we believe we can be ready for our pilot in two weeks and if we don't we believe we will be ready by next week. Can we adjust our plans with your team to accommodate that change?

`sitewide-content-accessibility-ia`

> Raquel Eisele
[Thursday Nov. 7 at 1:53 PM](https://dsva.slack.com/archives/C01K37HRUAH/p1731005598123579?thread_ts=1730990782.339339&cid=C01K37HRUAH)
>
> - Thank you so much for your response 
> - Sara (https://dsva.slack.com/team/U04NBMUH5C4)
> - **In response to your questions:** 
>     - **When are you looking to hold the Pilots?** 
>         - We would like to hold them later next week or early the week of the 18th. 
>         - I did start talking to Jamie about this.
>     - **What AT are you interested in testing?** 
>         - Desktop Screen Reader, Mobile Screen reader, Magnification/Zoom and Speech input Tech (Sire, Dragon).
>    - **What devices are you interested in testing?**
>        - We are interested in testing both mobile and desktop.
> - Thank you for sharing the Assistive Tech Pilot Guide!

cc: @coforma-jamie @mutiaadhocteam

**Comment by SarahKay8** [comment]
Pilot Notes

Links that serve as secondary actions should clearly convey their purpose. For example, “Remind me later” might not be sufficient for users who need more context. Consider labeling it as "Remind me later and return to the previous page". .	

Text and Link Clarity:
	•	The call-to-action phrases should be unambiguous. 

For example:
Instead of “Add your contact information,” you might specify, “Update your VA profile with contact information

  Your contact information has been updated page 

Suggestions for Improvement1.	Add Confirmation Feedback:
•	For the message “Your contact information has been updated,” ensure visual confirmation is supplemented by an accessible announcement, such as using aria-live pilote to dynamically inform screen reader users of success:

<div aria-live="polite">Your contact information has been updated.</div

 

Informational Context for “Appointment Reminders”:•	Issue: 

The listed items under “If you’re enrolled in VA health care…” are clear, but ensure context is provided for how and where users will receive this information (email, phone, etc.). This is critical for users relying on assistive technology.
	
Fix: Consider adding clarifying details, such as “We will send these updates via email or mail, depending on your preferences.”5.

**Comment by EvanAtCoforma** [comment]
# Pilot Notes
## Success Page

- Consider adding an `aria-live="polite"` to the success alert to ensure screen reader users are presented with this information upon page load.
- Pilot was unable to navigate the success page with the arrow keys while using a screen reader. Not sure if this is markup issue or something buggy that happened with the screen reader itself. Ensure markup is consistent with previous pages, and if so, could be a one-off issue.

## Error messages

- Interesting functionality with the error messages 
   - When there is no content in the required field, the error will say “you must provide a response”
   - Once content is added in, another message shows dynamically saying the number must have 10 digits
   - These errors are somewhat distracting as they’re alerted in the middle of the user typing content and the message could be missed
   - Pilot and co-pilot had issues with NVDA and JAWS screen readers with getting the error messages read correctly as they change dynamically.
   - NOTE: This is a component issue - nothing to do with the actual form

## Contact Information

For the contact information on the main page, we recommend making the content more specific to the kind of information being updated, as mentioned in the live audit notes as well. This should be done using visible text rather than any screen reader specific text (aria-label) as there are visual users who would appreciate this distinction as well.


cc: @raquelou
